android/app/src/main/java/com/chorestory/fragment/ChildFriendsFragment.java

```diff
@@ -1,20 +1,144 @@
+                disableViews();
+                String username = addFriendEditText.getText().toString();
+                if (username.trim().length() > 0) {
+                    addFriend(username);
+                } else {
+                    Toaster.showToast(getContext(), "Please enter a username");
+                }
+                enableViews();
+            }
+        });
+
         return view;
     }
+
+    private void addFriend(String username) {
+        String token = tokenHandler.getToken(getContext());
+        if (token != null) {
+            if (tokenHandler.isChildToken(token)) {
+                Call<AddFriendResponse> addFriendQuery = retrofitInference.add_friend(token, username);
+
+                addFriendQuery.enqueue(new Callback<AddFriendResponse>() {
+                    @Override
+                    public void onResponse(Call<AddFriendResponse> call, Response<AddFriendResponse> response) {
+                        if (response.isSuccessful() && response.body() != null && response.body().hasResponse()) {
+                            Toaster.showToast(getContext(), "Successfully added friend!");
+                            addFriendEditText.setText("");
+                            getChildren();
+                        } else {
+                            // TODO: friend doesn't exist
+                            Toaster.showToast(getContext(), "Friend not found");
+                        }
+                    }
+
+                    @Override
+                    public void onFailure(Call<AddFriendResponse> call, Throwable t) {
+                        // TODO: delete the token we have stored and redirect the user to the login page?
+                    }
+                });
+            } else {
+                // TODO: redirect to login page
+            }
+        }
+    }
+
+    private void getChildren() {
+        String token = tokenHandler.getToken(getContext());
+        if (token != null) {
+            if (tokenHandler.isChildToken(token)) {
+                Call<ClanChildrenResponse> getFriendsQuery = retrofitInference.list_friends(token);
+
+                getFriendsQuery.enqueue(new Callback<ClanChildrenResponse>() {
+                    @Override
+                    public void onResponse(Call<ClanChildrenResponse> call, Response<ClanChildrenResponse> response) {
+                        if (response.isSuccessful() && response.body() != null && response.body().hasResponse()) {
+                            childrenList = response.body().getChildren();
+                            childAdapter = new ChildRecyclerViewAdapter(childrenList);
+                            childRecyclerView.setAdapter(childAdapter);
+                        }
+                    }
+
+                    @Override
+                    public void onFailure(Call<ClanChildrenResponse> call, Throwable t) {
+                        // TODO: delete the token we have stored and redirect the user to the login page?
```

Comment:
What if request times out? Do we then sign people out for no reason?

Reply:
good point ill just add a toast with Internal error occurred.